### DISCUSSION TITLE
Use without /plain/ in url

### DISCUSSION THREAD

**Discussion by sca1235** [imgproxy/imgproxy#1027] [Q&A] [answered]
Hi looking to clean up the URL some and be able to resize without having /plain/

i.e. instead of `http://127.0.1.0:8080/resizing_type:fill/plain/12345.jpeg`

i would want `http://127.0.1.0:8080/resizing_type:fill/12345.jpeg`

i do not want to use encoded urls. also trying this with google storage bucket.

I thought maybe I could do by putting /plain/ in base url. `IMGPROXY_BASE_URL = /plain/gs://bucket222/` but unfortunately its not working saying Source image is unreachable

im sure its basic I'm just stuck trying to figure out what setting to do.

**Comment by DarthSim** [comment, marked as answer]
Hey!
imgproxy needs a way to separate base64-encoded URLs from plain ones. The `plain` section is a good token of a plain URL, as it is decoded to unreadable characters, so no base64-encoded URL can start with it. Anyway, what's wrong with it?

**Reply by sca1235** [reply]
Was just looking to clean up the urls some. To make it shorter for viewing and if someone copies and paste shorter so similar to past structures. I don't like plain urls :)

maybe a config so can always expect plain vs encoded. might be a minor security improvement too

I think i can figure rewrites with another thing like nginx but I really like the simplicity imgproxy provides right now.

**Reply by DarthSim** [reply]
It's not as simple as it may look at the first glance. imgproxy needs a way to determine where the processing options end and the source URL starts. It's simple in the case of base64-encoded URLs as they don't look like a processing option. But it may be complicated in the case of plain URLs. For example, `http:/` does look like a processing option to imgproxy.
Also, adding more ways to define the source URL increases the complexity of usage. I don't think the possibility of removing the `/plain/` section from the URL is worth it.

**Reply by sca1235** [reply]
Ok i see makes sense. 

What about option for shortening it? i.e. /p/ or /pn/

**Reply by DarthSim** [reply]
I don't think removing 3-4 characters from a URL is worth the change. If you so desperately want to shorten URLs this way, it's always possible to fork the repo.
